tests/cypress/integration/admin.test.js

```diff
@@ -0,0 +1,13 @@
+describe('Admin can login and make sure plugin is activated', () => {
+	it('Can activate plugin if it is deactivated', () => {
+		cy.login();
+		cy.visit('http://localhost:8889/wp-admin/plugins.php');
+		cy.get('body').then(($body) => {
+			if ($body.find('#deactivate-simple-podcasting').length > 0) {
+				cy.get('#deactivate-simple-podcasting').click();
+			}
+		});
```

Comment:
Doing conditional testing [is not recommended in Cypress](https://docs.cypress.io/guides/core-concepts/conditional-testing). It can lead to really brittle tests. For instance, if WordPress started rendering the plugin deactivation buttons with JavaScript, you might check for its presence before it's there. You can end up with weird race conditions where the tests fail on a slow machine, or fail on a fast machine.

Instead of doing conditional testing when you're unsure of the database's state before the test is run, **make it so** beforehand.

So, before you `cy.visit()` the plugins page, you could run this:

`cy.exec('npm run env run tests-cli wp plugin deactivate simple-podcasting')`

And then you you can just do `cy.get('#deactivate-simple-podcasting').click();`

Reply:
> cy.exec('npm run env run tests-cli wp plugin deactivate simple-podcasting')

This adds a dependency to the environment (access to the environment). Often it's better to be able to run e2e tests against a given URL without knowing how the environment is actually prepared (docker, local, VM...). This allows running the same suite of tests against multiple environments, comparison... It might not be a requirement here but it's one that can be important sometimes so I just wanted to raise it.

Reply:
Yep, that's a good thing to keep in mind. You could also abstract it into `cy.deactivatePlugin('simple-podcasting')` and handle environment differences there. I'm just wary of conditional code in UI tests, because it means you're going into the test with uncertainty about what you should be encountering, and repeat runs of the test could take different paths and make different assertions.